test/unit_tests/braket/experimental/autoqasm/test_parameters.py

```diff
@@ -0,0 +1,289 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""AutoQASM tests for parameter support."""
+
+import braket.experimental.autoqasm as aq
+from braket.circuits import FreeParameter
+from braket.default_simulator import StateVectorSimulator
+from braket.devices.local_simulator import LocalSimulator
+from braket.experimental.autoqasm import pulse
+from braket.experimental.autoqasm.instructions import cnot, cphaseshift, measure, ms, rx, rz
+from braket.tasks.local_quantum_task import LocalQuantumTask
+
+
+def _test_parametric_on_local_sim(program: aq.Program, inputs: dict[str, float]) -> None:
+    device = LocalSimulator(backend=StateVectorSimulator())
+    task = device.run(program, shots=10, inputs=inputs)
+    assert isinstance(task, LocalQuantumTask)
+    assert isinstance(task.result().measurements, dict)
+    return task.result().measurements
+
+
+@aq.main
+def simple_parametric():
+    rx(0, FreeParameter("theta"))
+    measure(0)
```

Comment:
I'm a bit confused by this ux over something like
```
@aq.main
def simple_parametric(theta):
    rx(0, theta)
    measure(0)
```
It seems like ^ is simpler and gives users a spot to designate inputs as ints etc

Reply:
I believe this already works with this implementation. User can then call either `simple_parametric(pi/2)` or `simple_parametric(FreeParameter("theta"))` depending on whether they want a fixed or parameterized program.